src/librustdoc/html/highlight.rs

```diff
@@ -442,10 +497,84 @@ fn exit_span(out: &mut Buffer) {
 /// ```
 /// The latter can be thought of as a shorthand for the former, which is more
 /// flexible.
-fn string<T: Display>(out: &mut Buffer, text: T, klass: Option<Class>) {
+fn string<T: Display>(
```

Comment:
The doc comment for this should be expanded

Reply:
I don't consider this resolved: this function is doing a _lot_ more now, and it should both be clear:

 - what it does
 -  what all the parameters are

Reply:
The function existed before and I extended its documentation. The new arguments are already explained in a few different places. I think this is a bit outside the scope of this PR.

Reply:
I'll try to extend it a bit more though (because more doc is always better).